mono/utils/atomic.h

```diff
@@ -464,4 +464,12 @@ extern void InterlockedWritePointer(volatile gpointer *dst, gpointer val);
 #define InterlockedAddP(p,add) InterlockedAdd64 ((volatile gint64*)p, (gint64)add)
 #endif
 
+/* The following functions cannot be found on any platform, and thus they can be declared without further existence checks */
+
+static inline gint32
+InterlockedSubtract (volatile gint32 *dest, gint32 sub)
+{
```

Comment:
`InterlockedAdd (location, -value);` will do the same thing you're trying to do with this function. I'd prefer that to introducing more functions here, since this header is sorta supposed to be as close to the Win32 Interlocked APIs as possible.

I do want to, At Some Point™, replace this whole header with an API that's closer to the C++11 atomics API (which includes subtract) since that's what the world is converging on (whether we like it or not).